src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,148 @@
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::RefrigeratorAlarm;
+using namespace chip::app::Clusters::RefrigeratorAlarm::Attributes;
+using namespace chip::DeviceLayer;
+using chip::Protocols::InteractionModel::Status;
+
+using namespace std;
+RefrigeratorAlarmServer RefrigeratorAlarmServer::instance;
+
+RefrigeratorAlarmServer & RefrigeratorAlarmServer::Instance()
+{
+    return instance;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::GetMaskValue(EndpointId endpoint, BitMask<AlarmMap> * mask)
+{
+    EmberAfStatus status = Attributes::Mask::Get(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: reading  mask %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::GetStateValue(EndpointId endpoint, BitMask<AlarmMap> * state)
+{
+    EmberAfStatus status = Attributes::State::Get(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: reading state %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state->Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::SetMaskValue(EndpointId endpoint, const BitMask<AlarmMap> & mask)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    status               = Attributes::Mask::Set(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: writing  mask %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: Mask ep%d value: %" PRIu32 "", endpoint, mask.Raw());
+
+    return status;
+}
+
+EmberAfStatus RefrigeratorAlarmServer::SetStateValue(EndpointId endpoint, BitMask<AlarmMap> state)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    BitMask<AlarmMap> currentState;
+
+    status = Attributes::State::Get(endpoint, &currentState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: reading  state %x", status);
+        return status;
+    }
+
+    status = Attributes::State::Set(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Refrigerator Alarm: ERR: writing  state %x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Refrigerator Alarm: State ep%d value: %" PRIu32 "", endpoint, state.Raw());
+
+    // Generate Notify evenet.
+    BitMask<AlarmMap> becameActive;
+    becameActive.SetRaw((~currentState.Raw()) & state.Raw());
```

Comment:
Why are you insisting on using bit-ops on Raw()?  How about:
```suggestion
    becameActive.Set(newState).Clear(currentState);
```